src/lakefs_spec/spec.py

```diff
@@ -352,21 +370,29 @@ def put(
         maxdepth=None,
         **kwargs,
     ):
+        repository, branch, resource = parse(rpath)
+
         super().put(
             lpath, rpath, recursive=recursive, callback=callback, maxdepth=maxdepth, **kwargs
         )
 
+        if self.create_branch_ok:
+            check_branch_exists_or_create(self.client,repository, branch)
+
```

Comment:
This should probably happen before the call to `put` - and if you use `parse` there, it should (as of now) come with a `rpath = self._strip_protocol(rpath)` before. (But we should change the parsing so that is not required.)

Reply:
Here I am unsure whether or not I adressed your comment with [these Changes](https://github.com/appliedAI-Initiative/lakefs-spec/pull/15/files#diff-f8235da430cb8c0f32975a91dcd263e02044b2621322a7e6c97182b53649ea1eR414). If not, please elaborate :)